### PR TITLE
Add rule for `com.intellij:annotations` and `org.jetbrains:annotations`

### LINKED ISSUE #39: Rule for `com.intellij:annotations`
It conflicts with `org.jetbrains:annotations`

### PR DESCRIPTION
**Pull request by boris-petrov**
Fixes #39 

cc @jjohannes 

I've no idea if what I did is correct, I'm not sure how to test it. :D You might add some info on building to the README. I also am not sure how to generate the `build.out` file for the two sample projects?

### CODE REVIEW

**Comment by jjohannes** [comment]
Thanks @boris-petrov.

Yes I should provide better instructions for contributing (#13 ... too much to do...)

For the tests, you can look at the test report and it will show the changed output. You can then copy/paste that into `build.out` (and double check if it is what you would expect).

The README will update automatically when running `check`.

**Comment by boris-petrov** [comment]
@jjohannes thanks for the help! I believe everything is fine now. But please double-check. :)

File: src/main/java/org/gradlex/javaecosystem/capabilities/rules/IntellijAnnotationsRile.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package org.gradlex.javaecosystem.capabilities.rules;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.api.artifacts.CacheableRule;
+import org.gradle.api.artifacts.ComponentMetadataContext;
+import org.gradle.api.artifacts.ComponentMetadataRule;
+
+@CacheableRule
+@NonNullApi
+public abstract class IntellijAnnotationsRile implements ComponentMetadataRule {
```
Comment (jjohannes) [inline src/main/java/org/gradlex/javaecosystem/capabilities/rules/IntellijAnnotationsRile.java:26]:
```suggestion
public abstract class IntellijAnnotationsRule implements ComponentMetadataRule {
```

small typo

**Review by jjohannes — CHANGES_REQUESTED** [review]

**Comment by boris-petrov** [comment]
Nice catch! Force-pushed.

**Review by jjohannes — APPROVED** [review]

**Pull request merged**